queue_garbage_collector/models/garbage_collector.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from datetime import timedelta
+
+from openerp import api, fields, models
+from openerp.addons.connector.queue.job import job
+from openerp.addons.connector.session import ConnectorSession
+
+
+@job
+def cron_checkup(session, model_name, in_queue_delta=5, started_delta=180):
+    """Fix jobs that are in a bad states
+    :param in_queue_delta: lookup time in minutes for jobs
+                            that are in enqueued state
+
+    :param started_delta: lookup time in minutes for jobs
+                            that are in enqueued state
+    """
+    obj = session.env[model_name]
+    obj.treat(
+        obj.get_jobs_to_requeue(
+            in_queue_delta=in_queue_delta,
+            started_delta=started_delta
+        )
+    )
+    return True
```

Comment:
Using a job for this will be an issue, because the main reason why we would want to requeue jobs stuck in started or enqueued is because they are stuck and block the queue: this job will likely never be executed. Better by executed directly by the cron

Reply:
maybe we can move it to a separate channel? Or does the queue get blocked completely?

Reply:
It depends: if a job is stuck in "started" in the root channel, it blocks other jobs.
It's not worth having this method in a job, keep it simple.